File: algorithm/src/main/java/baekjoon/p05086/Main.java
package baekjoon.p05086;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (true) {
            int a = sc.nextInt();
            int b = sc.nextInt();

            if (a == 0 && b == 0) {
                break;
            }

            System.out.println(solve(a, b));
        }
    }

    private static String solve(int a, int b) {
        if (a % b == 0) {
            return "multiple";
        } else if (b % a == 0) {
            return "factor";
        } else {
            return "neither";
        }
    }

}

File: algorithm/src/main/java/baekjoon/p05086/Main.kt
package baekjoon.p05086

fun main(args: Array<String>) {

    while (true) {
        val (a, b) = readLine()!!.split(" ").map { it.toInt() }

        if (a == 0 && b == 0) {
            break
        }

        println(solve(a, b))
    }
}

private fun solve(a: Int, b: Int): String {
    return when {
        (a % b == 0) -> "multiple"
        (b % a == 0) -> "factor"
        else -> "neither"
    }
}